src/test/java/tdl/s3/C_OnDemand_IncompleteFileUpload_AccTest.java

```diff
@@ -160,4 +162,49 @@ public void should_be_able_to_upload_failed_parts() throws Exception {
         assertTrue(remoteTestBucket.getObjectMetadata(fileName)
                 .getETag().startsWith(targetSyncFolder.getCompleteFileMD5(fileName)));
     }
+    
+    @Test
+    public void should_be_able_handle_empty_file() throws Exception {
```

Comment:
This test also covers `should_be_able_to_continue_incomplete_file_and_finalise` so we should keep only one.